_drafts/practical_zephyr_01_kconfig.md

```diff
@@ -0,0 +1,986 @@
+
+```bash
+$ rm -rf ../build
+$ west build --board nrf52840dk_nrf52840 -d ../build -- -DCONF_FILE=prj_release.conf
+```
+```
+Parsing /opt/nordic/ncs/v2.4.0/zephyr/Kconfig
+Loaded configuration '/opt/nordic/ncs/v2.4.0/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
+Merged configuration '/path/to/01_kconfig/prj_release.conf'
+Configuration saved to '/path/to/build/zephyr/.config'
+```
+
+Well, this is awkward: Even though we provided a board-specific `Kconfig` fragment it is ignored as soon as we switch to our alternative `prj_release.conf` configuration. This is, however, [working as intended](https://docs.zephyrproject.org/latest/build/kconfig/setting.html#the-initial-configuration): When using alternative `Kconfig` files in the format `prj_<build>.conf`, Zephyr looks for a board fragment matching the file name `boards/<board>_<build>.conf`.
+
+In my case, I need to create the file `boards/nrf52840dk_nrf52840_release.conf`. Let's get rid of the warning for the Zephyr library `drivers__console` by disabling the `CONSOLE` entirely in our board's release configuration:
+
+```bash
+tree --charset=utf-8 --dirsfirst
+.
+├── boards
+│   ├── nrf52840dk_nrf52840.conf
+│   └── nrf52840dk_nrf52840_release.conf
+├── src
+│   └── main.c
+├── CMakeLists.txt
+├── prj.conf
+└── prj_release.conf
+
+$ cat boards/nrf52840dk_nrf52840_release.conf
+```
+```conf
+CONFIG_CONSOLE=n
+CONFIG_UART_CONSOLE=n
+```
+
+Now, when we rebuild the project we see that our board fragment is merged into the `Kconfig` configuration, and no warning is issued by `west build`:
+
+```bash
+$ rm -rf ../build
+$ west build --board nrf52840dk_nrf52840 -d ../build -- -DCONF_FILE=prj_release.conf
+```
+```
+Parsing /opt/nordic/ncs/v2.4.0/zephyr/Kconfig
+Loaded configuration '/opt/nordic/ncs/v2.4.0/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
+Merged configuration '/path/to/01_kconfig/prj_release.conf'
+Merged configuration '/path/to/01_kconfig/boards/nrf52840dk_nrf52840_release.conf'
+Configuration saved to '/path/to/build/zephyr/.config'
+```
+
+
+### Extra configuration files
+
+There is one more option to modify the _Kconfig_, listed among the [important build system variables in Zephyr's official documentation](https://docs.zephyrproject.org/latest/develop/application/index.html#important-build-system-variables): `EXTRA_CONF_FILE`. This build system variable accepts one or more additional _Kconfig_ fragments. This option can be useful, e.g., to specify additional configuration options used by multiple build types (normal builds, "release" builds, "debug" builds) in a separate fragment.
+
+> **Note:** Since [Zephyr 3.4.0](https://docs.zephyrproject.org/latest/releases/release-notes-3.4.html) the `EXTRA_CONF_FILE` build system variable replaces the deprecated variable `OVERLAY_CONFIG`.
+
+Let's add two _Kconfig_ fragments `extra0.conf` and `extra1.conf`:
```

Comment:
I would give these more descriptive names like `no-debug.conf` and `no-gpio.conf`

Reply:
Hmm I kind of like the naming for the reference example since it directly shows that the files are "extra" config files (and the number in the file name emphasizes the order). I'll mention your names in the article hoping that no one will follow my naming for an actual project.